app/controllers/tags_controller.rb

```diff
@@ -1,22 +1,16 @@
 class TagsController < ApplicationController
-
   def index
-    @tags =  Tag.search(params[:q])
-  end
-
-  def popular
-    render json: Tag.get_tags_that_are(:popular), status: 200
-  end
-
-  def recent
-    render json: Tag.get_tags_that_are(:recent), status: 200
+    render partial: 'tag_properties', locals: { tags: Tag.search(params[:q])}
```

Comment:
Is there a reason we can't use the `_tags` partial we already have or did you rename it to `tag_properties`?

My second suggestion would be, do you think it will be more readable and more conventional to set the `@tags` instance variable and then just render partial `tag_properties`, instead of passing `locals` hash which seem to be more low-level.  😨